theories/algebra/StdBigop.ec

```diff
@@ -86,6 +86,11 @@ lemma nosmt big_constz (P : 'a -> bool) x s:
   BIA.big P (fun i => x) s = x * (count P s).
 proof. by rewrite BIA.sumr_const -IntID.intmulz. qed.
 
+lemma nosmt big_constz_val (P : 'a -> bool) F x s:
+     (forall i , P i => F i = x)
+  => BIA.big P F s = x * (count P s).
+proof. rewrite -big_constz &(BIA.eq_bigr). qed.
```

Comment:
I am not sure this is necessary. If we start adding "general pattern" lemmas, where do we stop?

Reply:
Alright sure, but would this lemma then not be preferred over `big_constz`? This because this lemma should be applicable whenever `big_constz` is applicable; and whenever applying this lemma on a case where one could apply `big_constz`,  the goal generated by the precondition should immediately be solved by `trivial` as F would be instantiated with the function that ignores its input and simply returns `x`.

Reply:
Not necessarily: `big_constz` is safe (e.g. with `?`) and only matches expressions were applying it is useful. Your lemma applies to every instance of `BIA.big`, possibly generating an unprovable side condition.

Reply:
That's fair; I would also never have written this lemma if I could efficiently rewrite functions that are operator arguments (and hence be able to quickly prove that my function F is equivalent to the function used in `big_constz`).  However, since this is not possible (as far as I know?), and I had several cases of this in a row, I decided to prove this lemma. Not sure if it makes sense to keep it in the standard library, I'll let you make that call.

Reply:
Here is a "forward" proof of your lemma:
```
by move => ?; rewrite (BIA.eq_bigr _ _ (fun _ => x)) 1:/# big_constz.
```
This is the usual pattern to get a bigop  into the required shape.

I'm not against this kind of lemma. But if we add them, this should be a separate PR with decent coverage and a good naming convention.

Reply:
Sure, that sounds reasonable. I will remove them from this PR then.